lib/utils/queryable.js

```diff
@@ -119,14 +144,33 @@ const setter = ({ data, key, value, force }) => {
   // ['foo', 'bar', 'baz'] -> { foo: { bar: { baz:  {} } }
   const keys = parseKeys(key)
   const setKeys = (_data, _key) => {
-    // handles array indexes, making sure the new array is created if
-    // missing and properly casting the index to a number
-    const maybeIndex = Number(_key)
-    if (!Number.isNaN(maybeIndex)) {
+    // handles array indexes, converting valid integers to numbers,
+    // note that occurences of Symbol(append) will throw,
+    // so we just ignore these for now
+    let maybeIndex = Number.NaN
+    try {
+      maybeIndex = Number(_key)
+    } catch (err) {}
+    if (!Number.isNaN(maybeIndex))
       _key = maybeIndex
-      if (!Object.keys(_data).length)
-        _data = []
-    }
+
+    // creates new array in case key is an index
+    // and the array obj is not yet defined
+    const keyIsAnArrayIndex = _key === maybeIndex || _key === _append
+    const dataHasNoItems = !Object.keys(_data).length
+    if (keyIsAnArrayIndex && dataHasNoItems)
```

Comment:
```suggestion
    if (keyIsAnArrayIndex && dataHasNoItems && !Array.isArray(_data))
```
? this might be unnecessary but it is a bit more explicit.

also, what happens if `data` is a non-object and non-array? like `foo=3&foo[0]=3&foo.bar=3`?

Reply:
if not force mode, the first problem encountered is going to error out:

```
$ npm pkg set foo=3 foo[0]=3 foo.bar=3
npm ERR! code EOVERRIDEVALUE
npm ERR! Property foo[0] already has a value in place.

npm ERR! A complete log of this run can be found in:
npm ERR!     $HOME/.npm/_logs/2021-07-13T21_36_11_855Z-debug.log
```

if forcing, it gets overrwritten as expected:

```
$ npm pkg set foo=3 foo[0]=3 foo.bar=3 --force
npm WARN using --force Recommended protections disabled.
```
```patch
$ git diff
diff --git a/package.json b/package.json
index 38c21cf..6409dd1 100644
--- a/package.json
+++ b/package.json
@@ -6,5 +6,9 @@
   },
   "dependencies": {
     "@ruyadorno/slow-lifecycle-script": "^1.0.3"
+  },
+  "foo": {
+    "0": "3",
+    "bar": "3"
   }
 }
```

Reply:
"Property foo[0] already has a value in place." isn't accurate there tho - the problem is that "Property foo already exists and is not an Array or an Object"

Reply:
alright, I'll give you this one:

```
$ npm pkg set foo=3 foo[0]=3 foo.bar=3
npm ERR! code EOVERRIDEVALUE
npm ERR! Property foo already exists and is not an Array or Object.

npm ERR! A complete log of this run can be found in:
npm ERR!     $HOME/.npm/_logs/2021-07-13T21_53_58_011Z-debug.log
```